### DISCUSSION TITLE
simplefix on binance

### DISCUSSION THREAD

**Discussion by abstractumx** [da4089/simplefix#43] [Q&A] [answered]
hi, 

it is possible to use fix protocol wrapped into python code to binance exchange?

thanks

**Comment by jvirtanen** [comment, marked as answer]
As far as I know, Binance does not have a native FIX API, so you cannot use e.g. simplefix to connect to it.

**Reply by mecanda12** [reply]
@jvirtanen How can use simple Fix library to connect client or feed bundle? any example you can share?

**Reply by da4089** [reply]
> @jvirtanen How can use simple Fix library to connect client or feed bundle? any example you can share?

What are you trying to connect to?  Is it a FIX server?

**Reply by mecanda12** [reply]
@da4089 I am trying to connect trading engine directly? is that possible? or do I need FIX server?

**Reply by da4089** [reply]
If you are trying to connect to a venue that offers a FIX API, you can use this library (or any other FIX implementation).  If you're trying to connect to Binance, then this is the wrong place to be asking: no-one here is a Binance expert.

Assuming you want to connect to a FIX venue, then _simplefix_ is one way of doing that from Python.  But ... it is quite limited in scope -- you will need to deal with everything other than encoding and decoding FIX messages yourself.  If you want a more complete FIX implementation, you should look at [QuickFIX](https://github.com/quickfix/quickfix) or [WTFIX](https://github.com/jcass77/WTFIX) (both available from PyPI).

If you want to proceed with _simplefix_, you'll need to decide how you want to structure your networking (callbacks, threads, asyncio), and how you want to persist your sequence numbers and message history (files? database?).  With those choices in hand, you can establish a TCP (or SSL) connection to the venue, and use _simplefix_ to encode and decode the FIX messages.

Hope this helps?

**Reply by mecanda12** [reply]
I prefer simplefix and want to procced with that. I will be using this library for testing purpose only for regression. I do not think need history and database. All I need establish the Network, that is where I need help.
My understanding is : Regression Test suite -----> Simplefix library-------> Trading engine  

By the way,  Do I need to set up a fix server separately to connect with trading engine?

**Reply by da4089** [reply]
Migrated the discussion on regression testing to #55

**Comment by abstractumx** [comment]
ok thanks

Il giorno lun 10 ott 2022 alle ore 18:12 Jussi Virtanen <
***@***.***> ha scritto:

**Comment by da4089** [comment]
Thanks @jvirtanen -- I've had a quick look at Binance's web site, and there's no sign of FIX support.  They appear to have a REST API for orders, and WebSockets for market data.  They also have API implementations for various languages, including Python.

You could start looking at https://binance-docs.github.io/apidocs/spot/en/ for details.

**Comment by abstractumx** [comment]
Hi,

thanks for letting me know valid alternatives for Binance exchange.

My questions comes up when I started to check performance in terms of
latency and I’ve found out that fix protocol was one of the most affordable
in terms of performance overall. That’s why my question came up to your
repo.

Based on your message the best practice is to get data using 2 Types of
APIs.

1) rest api for orders injections (hope can I set custom orders / scalable
orders)
2) websocket api for data feed and price updates

Am I correct? Yes, I prefer to use python. Alternatively c++ could be
useful but only for HFT, but is not my actual target. I don’t have the
right infrastructure in terms of connections and machines to support HFT.

Thanks
Have a great day.


Il giorno mar 11 ott 2022 alle 14:44 David Arnold ***@***.***>
ha scritto:

**Comment by da4089** [comment]
I have no experience nor expertise with the Binance APIs -- I simply looked at their website, as discovered by a search engine, and reported what I found.  Yes -- it appears that you need to use REST for orders and WebSockets for quotes.  This is pretty common for crypto venues.  Python should be fine for both those protocols.

**Reply by abstractumx** [reply]
Perfect, thank you so much!

Best, 
A.

**Comment by rapmd73** [comment]
> hi,
> 
> it is possible to use fix protocol wrapped into python code to binance exchange?
> 
> thanks

For connecting to Binance, you may find this an easier route,

https://github.com/rapmd73/JackrabbitRelay/wiki
